File: seed/csharp-model/examples/src/SeedExamples/Types/ResponseType.cs
using System.Text.Json.Serialization;
using OneOf;
using SeedExamples.Core;

namespace SeedExamples;

public record ResponseType
{
    [JsonPropertyName("type")]
    public required OneOf<BasicType, ComplexType> Type { get; set; }

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/php-model/examples/src/Types/ResponseType.php
<?php

namespace Seed\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\BasicType;
use Seed\ComplexType;
use Seed\Core\Json\JsonProperty;

class ResponseType extends JsonSerializableType
{
    /**
     * @var (
     *    value-of<BasicType>
     *   |value-of<ComplexType>
     * ) $type
     */
    #[JsonProperty('type')]
    public string $type;

    /**
     * @param array{
     *   type: (
     *    value-of<BasicType>
     *   |value-of<ComplexType>
     * ),
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->type = $values['type'];
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}
